src/api/app/models/package.rb

```diff
@@ -308,12 +308,14 @@ def is_locked?
     project.is_locked?
   end
 
+  def master_product_object
+    # test _product permissions if any other _product: subcontainer is used and _product exists
+    product_object = project.packages.find_by(name: "_product") if name =~ /\A_product:\w[-+\w\.]*\z/
```

Comment:
I'm not too familiar with the context of this change, but is there a way that this can be done without a regex match on the package name? It assumes that all packages follow this naming convention. What happens if somebody changes the way these packages are named and doesn't know that they need to update this?

I think  a better option is to add some new columns to the packages tables so we can have associations which handle this like bjoern suggested.